packages/shared/topology/sidebar/deployments/DeploymentResources.tsx

```diff
@@ -0,0 +1,164 @@
+  WatchK8sResource,
+} from '@openshift-console/dynamic-plugin-sdk';
+import * as _ from 'lodash-es';
+import { isPodInDeployment } from '../../utils/resource';
+import { PodsOverviewList, PodWithMetricsKind } from '../common/PodList';
+import './node-resources.scss';
+
+export const odfStatefulSetResource: WatchK8sResource = {
+  isList: true,
+  kind: referenceForModel(StatefulSetModel),
+  namespaced: true,
+  namespace: 'openshift-storage',
+};
+
+export const odfReplicaSetResource: WatchK8sResource = {
+  isList: true,
+  kind: referenceForModel(ReplicaSetModel),
+  namespaced: true,
+  namespace: 'openshift-storage',
+};
+
+export const odfDaemonSetResource: WatchK8sResource = {
+  isList: true,
+  kind: referenceForModel(DaemonSetModel),
+  namespaced: true,
+  namespace: 'openshift-storage',
+};
+
+const getPodMetric = (
+  metrics: PrometheusResponse['data']['result'],
+  podName
+): number => {
+  const podMetric = _.findIndex(
+    metrics,
+    (metric: PrometheusResult) => metric?.metric?.pod === podName
+  );
+  return podMetric > -1 ? Number(metrics[podMetric]['value'][1]) : NaN;
+};
+
+type DeploymentResourceProps = {
+  deployment: DeploymentKind;
+};
+
+export const DeploymentResources: React.FC<DeploymentResourceProps> = ({
+  deployment,
+}) => {
+  const { t } = useCustomTranslation();
+  const [pods, podsLoaded, podsLoadError] = useK8sWatchResource<
+    PodWithMetricsKind[]
+  >({
+    kind: PodModel.kind,
+    isList: true,
+    namespaced: true,
+    namespace: CEPH_STORAGE_NAMESPACE,
+  });
+
+  const [statefulSets, statefulSetLoaded, statefulSetError] =
+    useK8sWatchResource<K8sResourceCommon[]>(odfStatefulSetResource);
+
+  const [replicaSets, replicaSetsLoaded, replicaSetsError] =
+    useK8sWatchResource<K8sResourceCommon[]>(odfReplicaSetResource);
+
+  const [daemonSets, daemonSetsLoaded, daemonSetError] =
+    useK8sWatchResource<K8sResourceCommon[]>(odfDaemonSetResource);
+
+  const memoizedStatefulSets = useDeepCompareMemoize(statefulSets, true);
+  const memoizedReplicaSets = useDeepCompareMemoize(replicaSets, true);
+  const memoizedDaemonSets = useDeepCompareMemoize(daemonSets, true);
+
+  const filteredPods = React.useMemo(() => {
+    const resources = [
+      ...memoizedReplicaSets,
+      ...memoizedDaemonSets,
+      ...memoizedStatefulSets,
+      deployment,
+    ];
+    const loaded =
+      statefulSetLoaded && replicaSetsLoaded && daemonSetsLoaded && podsLoaded;
+    const loadError =
+      statefulSetError || replicaSetsError || daemonSetError || podsLoadError;
+    return loaded && !loadError && !_.isEmpty(pods)
+      ? pods.filter((pod) => isPodInDeployment(pod, ...resources))
+      : [];
+  }, [
+    daemonSetError,
+    daemonSetsLoaded,
+    deployment,
+    memoizedDaemonSets,
+    memoizedReplicaSets,
+    memoizedStatefulSets,
+    pods,
+    podsLoadError,
+    podsLoaded,
+    replicaSetsError,
+    replicaSetsLoaded,
+    statefulSetError,
+    statefulSetLoaded,
+  ]);
```

Comment:
lesser variables for React to compare...
```suggestion
    const loaded =
      statefulSetLoaded && replicaSetsLoaded && daemonSetsLoaded && podsLoaded;
    const loadError =
      statefulSetError || replicaSetsError || daemonSetError || podsLoadError;
  const filteredPods = React.useMemo(() => {
    const resources = [
      ...memoizedReplicaSets,
      ...memoizedDaemonSets,
      ...memoizedStatefulSets,
      deployment,
    ];
    return loaded && !loadError && !_.isEmpty(pods)
      ? pods.filter((pod) => isPodInDeployment(pod, ...resources))
      : [];
  }, [
    loaded,
    loadError,
    deployment,
    memoizedDaemonSets,
    memoizedReplicaSets,
    memoizedStatefulSets,
    pods,
  ]);
```